install/kubernetes/cilium/values.yaml.tmpl

```diff
@@ -650,6 +650,19 @@ enableK8sEventHandover: false
 # -- Enable CiliumEndpointSlice feature.
 enableCiliumEndpointSlice: false
 
+envoyConfig:
+  # -- Enable envoy config CRD
+  # Cilium Envoy Config CRD can also be implicitly enabled by other options.
+  enabled: false
+
+  # -- SecretsNamespace is the namespace in which envoy SDS will retrieve secrets from.
+  secretsNamespace:
+    # -- Create secrets namespace for CiliumEnvoyConfig CRDs.
+    create: true
```

Comment:
Should this also be `~` -- true if envoyConfig evaluates to true?

Reply:
We only ever test if this is `true` or not, so setting this as nil serves no purpose and adds an unneeded dimension to understanding the chart. It is simpler if this is always a well defined boolean.

Reply:
Nil has two main uses:

1. When there is any logic to determine the default value. This allows us to determine the default programmatically, while still allowing the end-user to override in either direction.
2. When the default value is and should be empty, but we would like to document the presence of a variable. Having a nil value means the docblock is consumed by tooling.

It not related directly to the agent.